runsc/container/container.go

```diff
@@ -1894,3 +1927,68 @@ func nvproxySetupAfterGoferUserns(spec *specs.Spec, conf *config.Config, goferCm
 		return nil
 	}, nil
 }
+
+// detectRootfsType detects the rootfs filesystem type.
+func detectRootfsType(spec *specs.Spec, conf *config.Config) string {
+	if _, ok := spec.Annotations[erofsRootfsAnnotation]; !ok {
+		return gofer.Name
+	}
+
+	for _, m := range spec.Mounts {
+		// Fallback to gofer if there is any gofer mount requested.
+		if specutils.IsGoferMount(m) {
+			return gofer.Name
+		}
+	}
```

Comment:
Why can't we use EROFS with gofer mounts? bind mounts are pretty common.

It seems there are 2 orthogonal ideas here:
1. When to start gofer? When at least 1 gofer mount exists OR GPU functionality is requested. A gofer mount is (rootfs + bind mounts).
2. When to use EROFS rootfs? When the annotation is set.

Reply:
Ah we require a rootfs to create bind mounts in the gofer: https://cs.opensource.google/gvisor/gvisor/+/master:runsc/cmd/gofer.go;l=341-406;drc=fb487fbb2fa1b2596d38633b1fedb2f2348bb711.

When EROFS is being used, we need to fake the rootfs (i.e. create a tmp dir in gofer and treat that as rootfs and create all bind mounts inside that and pivot_root into that).

Reply:
Makes sense. I was treating EROFS rootfs as goferless to some extent, so when gofer process needs to be launched, it will just fallback to gofer. I will see how to make them coexist.